### DISCUSSION TITLE
POI search (online)

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#375] [Q&A] [answered]
We can also have an online search for POI, so the POI will be the most _up-to-date_ in:

- Nearby POI in navigation
- Search screen:
  - Search POI categories
  - Search by POI name or type

Offline POI search has priority. Online POI search is used if offline POI search has not found results.

<img src="https://github.com/devemux86/cruiser/assets/3484020/881310dd-30ff-44af-afe5-cf7ec124c966" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/e78ebfee-23c9-41e9-b9d4-69182aed2b0a" width="50%"/>

**Comment by devemux86** [comment, marked as answer]
POI search, online together with offline, is available in [Cruiser 3.0.35](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

I didn't add any extra options to avoid complicating the settings.
Let's see how it works and if you need anything more.

**Comment by DolomitesTom** [comment]
![Screenshot_20231114-120936_Cruiser](https://github.com/devemux86/cruiser/assets/121371629/2ca79d62-884c-4c37-8461-7e8f07ff8023)

Have an actual example where I don't find valid data.

I have booked something at booking.com, copy the address from there and try to search: 
Chemin Combe La Noire 7, 2350, Schweiz

Something very far away is found, when I click it then the error message appears: location outside of map.
When I search on google maps, it finds it within a second.

![Screenshot_20231114-121549_Maps](https://github.com/devemux86/cruiser/assets/121371629/d483140a-9a2d-4d75-9410-dfe04c22b02b)

When I then copy the plus code, then Cruiser also finds it pretty quick, but not with the address or part of the address.

**Reply by devemux86** [reply]
@DolomitesTom thanks the report, this is a nice example.

The "7, 2350" confuses the search engine, thinking its coordinates and stops early the search.
We can certainly improve this, it will be available in the next update.

**Reply by devemux86** [reply]
@DolomitesTom these should be resolved in [Cruiser 3.0.36](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
Also OpenRouteService should now find more results.

**Comment by umpfmpf** [comment]
The search has now become a lot faster! 
But: what search should I use? "GraphHopper" "MapQuest" or "OpenRouteService"?

**Reply by devemux86** [reply]
OpenRouteService is the default option.

They use different algorithms, so they may show different results.
Better try all of them and see how they respond in your searches.

These services are mostly useful to find locations, addresses, etc.
Like above example "Chemin Combe La Noire 7, 2350, Schweiz".

**Reply by DolomitesTom** [reply]
> Like above example "Chemin Combe La Noire 7, 2350, Schweiz".

I tested it with all 3 search engine with 3.0.36 and it takes very long. It finds the location, but it takes more than 1 minute. How can this be speeded up a bit? Or is it always a compromise what is searched first? I have activated Austria, Belgium, France, Germany, Luxemburg and Swiss POI. Is this the reason why it takes that long?

**Reply by devemux86** [reply]
@DolomitesTom it shows the result very quickly.

What have you selected in "Settings | Search | Search"?

How many POI files you have activated in downloads?
How many favorites and route / tracks do you have?

Search is universal, it tries to find a text in all possible app places (if activated).
Only coordinates, plus codes and what3words can be recognized and return immediately.

**Reply by devemux86** [reply]
> I have activated Austria, Belgium, France, Germany, Luxemburg and Swiss POI. Is this the reason why it takes that long?

Probably, can you try deactivating some POI and make some tests?

**Reply by umpfmpf** [reply]
If I click on the address in the text program and then go to open with Cruiser, he will find the address. 
He doesn't find them in Cruiser's internal search... 
I only activated Germany Pois

**Reply by devemux86** [reply]
@umpfmpf what address?
What do you click, a link? Can you write the link?

Offline POI contain points of interests (with their addresses), not all location addresses.
To search addresses of locations the online search is needed.

**Comment by umpfmpf** [comment]
I have checked all search options. 
And when I enter the address in the search line and then mark it, I can choose what I want to do with it. I then select Cruiser as the navigation program and the address is displayed.

![Screenshot_20231115_134050_Cruiser](https://github.com/devemux86/cruiser/assets/105640513/4666bda2-f3f6-4e41-b57a-913e458d4619)
![Screenshot_20231115_134108_Cruiser](https://github.com/devemux86/cruiser/assets/105640513/a7b1639b-bb61-4d41-a15b-7d86d81da829)
![Screenshot_20231115_134136_Cruiser](https://github.com/devemux86/cruiser/assets/105640513/3b733221-002c-492b-8b85-8881571e1804)

**Comment by DolomitesTom** [comment]
had all search settings active... deactivated now tracks and some POIs. That makes it much quicker.

**Reply by devemux86** [reply]
Offline search is already optimized, by me and other developers ([Orux](https://github.com/mapsforge/mapsforge/issues/1443)).

However Germany and France databases contain over 10 million records!
A lot of (mobile) work, especially if don't zoom enough to search in specific regions.

So it is recommended to keep activated only the downloaded poi you need.
There are more optimizations in my mind, I need to test them in the future.

**Comment by DolomitesTom** [comment]
The search engine is now much better than before.

**Reply by umpfmpf** [reply]
That is true! 
So far, everything I need has actually been found and this much faster (apart from one address in Switzerland). 
Just a good job!!

**Comment by devemux86** [comment]
If something cannot found, please report it so we can check it.

You can also make search tests on the websites of the services:
- [OpenRouteService](https://maps.openrouteservice.org)
- [GraphHopper](https://graphhopper.com/maps)
- [MapQuest](https://www.mapquest.com)
